pkg/utils/kubeclient/job.go

```diff
@@ -39,3 +42,27 @@ func GetJob(client client.Client, name, namespace string) (*v1.Job, error) {
 func UpdateJob(client client.Client, job *v1.Job) error {
 	return client.Update(context.TODO(), job)
 }
+
+// GetSucceedPodForJob get the first finished pod for the job, if no succeed pod, return nil with no error.
+func GetSucceedPodForJob(c client.Client, job *v1.Job) (*corev1.Pod, error) {
+	var podList corev1.PodList
+	selector, err := metav1.LabelSelectorAsSelector(job.Spec.Selector)
+	if err != nil {
+		return nil, fmt.Errorf("error converting Job %s in namespace %s selector: %v", job.Name, job.Namespace, err)
+	}
+	err = c.List(context.TODO(), &podList, &client.ListOptions{
```

Comment:
Is it possible not to List pods? Listing pods in informer's cache turns out to be a heavy operation if there are a lot of pods in one namespace. In most case here, we try to find one succeed job pod but have to check all the pods in the same namespace, this may need large amount of CPU. WDYT @cheyang @xliuqq

Reply:
will add cache filter and index in future.